packages/flutter/lib/src/material/switch.dart

```diff
@@ -758,13 +808,17 @@ class _RenderSwitch extends RenderToggleable {
         break;
     }
 
-    final Color trackColor = isEnabled
+    final Color trackColor = isEnabled || useActiveColorInDisabledState
       ? Color.lerp(inactiveTrackColor, activeTrackColor, currentValue)!
       : inactiveTrackColor;
 
-    final Color thumbColor = isEnabled
+    final Color lerpedThumbColor = isEnabled || useActiveColorInDisabledState
       ? Color.lerp(inactiveColor, activeColor, currentValue)!
       : inactiveColor;
+    // Blend the thumb color against a `surfaceColor` background in case the
+    // thumbColor is not opaque. This way we do not see through the thumb to the
+    // track underneath.
+    final Color thumbColor = Color.alphaBlend(lerpedThumbColor, surfaceColor);
```

Comment:
Is it possible that this is a breaking change?

Reply:
I think it technically could be, but I don't believe it is considered one because none of the customer/internal tests fail. 

It would only break if someone was passing in a non-opaque value to the thumb color. I don't think people do that because 1) Before now, you couldn't customize disabled color (which is when opacity is typically used), and 2) Before now, if you passed in a non-opaque color, you would see the track underneath, which is probably not preferred.